subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/NoOutputsArtifactState.java

```diff
@@ -98,6 +99,11 @@ public IncrementalTaskInputs getInputChanges() {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public Iterable<? extends FileCollectionFingerprint> getInputFileFingerprints() {
+        throw new UnsupportedOperationException();
+    }
+
```

Comment:
I wish we'd be able to get rid of this thing---throwing `UOE` all over the place is hardly a good way of doing things.

Reply:
👍 
I had a quick look what would be necessary. We would need a second way through our chain of executers: https://github.com/gradle/gradle/blob/7f0f09d7cf52c81eb440865e66b30dc66cf93654/subprojects/core/src/main/java/org/gradle/internal/service/scopes/TaskExecutionServices.java#L130-L160

We would need to fork in `ResolveTaskArtifactStateTaskExecuter` - instead of creating a `TaskArtifactState` we would go a different route:
- without `CleanupStaleOutputsExecuter`
- with `FinalizeInputFilePropertiesTaskExecuter`
- a simpler version of `SkipEmptySourceFilesTaskExecuter`
- `ValidatingTaskExecuter`
- a simpler version of `ResolveBuildCacheKeyExecuter` - always returning an invalid build cache key
- a simpler version of `ResolveTaskOutputCachingStateExecuter`, always returning `NO_OUTPUTS_DECLARED` as a reason
- instead of `SkipUpToDateTaskExecuter` we would set a fixed out-of-date reason
- without `SkipCachedTaskExecuter`
- without `OutputDirectoryCreatingTaskExecuter`
- without `SnapshotAfterExecutionTaskExecuter`
- `TimeoutTaskExecuter`, `ActionEventFiringTaskExecuter` and `ExecuteActionsTaskExecuter` would remain unchanged.

With our current structure for the executers, this is not very easy to do. I guess we need to do something about how we use the executers when we introduce caching and up-to-date checks for artifact transforms. We can come back to the problem when we get there.